### PR TITLE
refactor: use region-helper

### PR DESCRIPTION
**Pull request by eunjae-lee**
## Pull request type

Jira Link: [EXT-2185](https://storyblok.atlassian.net/browse/EXT-2122)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Other (please describe):

## How to test this PR

Nothing changes. It should pass all the existing tests.

## What is the new behavior?

This PR includes the new `@storyblok/region-helper` library to replace the existing region-related code in this repository. The purpose is to centralize the logic into a single library, and reuse it everywhere (to ease the situation where we add new regions).

You can refer to the internal `#product-regions-configuration` channel or reach out to me for more information.

Basically [the library's code](https://github.com/storyblok/region-helper) is copied from this repository :)

Also, I've ran `npm run build` locally, and confirmed that the `region-helper` was correctly bundled into the final output.

[EXT-2185]: https://storyblok.atlassian.net/browse/EXT-2185?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by eunjae-lee** [comment]
I think the CI is not working ... because I'm an external contributor? Let me know if I should do something else :)

**Review by ademarCardoso — APPROVED** [review]
The code it's working great, i will ask to my QA review for us and then we can merge this, Thanks @eunjae-lee .

**Comment by eunjae-lee** [comment]
just added a commit to upgrade region-helper to 0.2.0, but nothing really changes :)

**Comment by BibiSebi** [comment]
Hey @ademarCardoso, can I support you here in any way (like finding or assigning a QA)? :) @eunjae-lee will be out of the office for the rest of the week so I will be glad to help if anything is needed.

**Comment by BibiSebi** [comment]
@eunjae-lee @ademarCardoso I added some changes to make sure that it is still possible to switch between http and https.

**Review by ademarCardoso — APPROVED** [review]
Hi @BibiSebi I've reviewed it again and also looked at the code once more in the region-helper repo, and everything looks perfect to me 🚀

File: src/index.ts [hunk 1]
```diff
@@ -87,13 +88,12 @@ class Storyblok {
 		let endpoint = config.endpoint || pEndpoint
 
 		if (!endpoint) {
-			const getRegion = new SbHelpers().getRegionURL
 			const protocol = config.https === false ? 'http' : 'https'
```
[the remark this answers is not in the document]
Comment (ademarCardoso) [reply src/index.ts:91]: I believe the protocol differences are to test environments that are still under development, but we normally only use https, but we need to keep http working for backward compatibility. However, with the code that's currently in the region-helper, I don't think we'll have any problems.

File: package.json [hunk 2]
```diff
@@ -120,5 +120,8 @@
 		"tabWidth": 2,
 		"trailingComma": "es5",
 		"useTabs": true
+	},
```
Comment (ts-storyblok) [inline package.json:123]: My only question about this feature is that the js client no longer uses external libraries. Community asked to remove and we did. Please check with the decision makers for this project to give you their green light before you release this version.

Comment (ademarCardoso) [reply package.json:123]: @BibiSebi As Thiago said, the focus of `storyblok-js-client` is to have no external dependencies in order to be as lightweight as possible. What do you think ?

**Comment by eunjae-lee** [comment]
Hi @ademarCardoso and @ts-storyblok ,
I was away last week attending a conference, and now I'm back.

I totally understand your concern. It's a good strategy not including third party dependencies that could increase the bundlesize and have potential security risk, or uncontrollable situations, etc.

However, this region-helper is a tiny library maintained by ourselves, and totally treeshakable. And I have confirmed that it actually hasn't increase the bundle size in this pull request.

For example, when I created this pull request, I created this branch based on the commit 3f9fb78, and if you

(A)
```
git checkout 3f9fb78
npm install
npm build
```

(B)
```
git checkout refactor/EXT-2185
npm install && npm build
```

We can see introducing this library didn't have any impact on the bundle size (the difference between (A) and (B) here)

| File               | Size - (A) | Size - (B) | Gzip Size - (A) | Gzip Size - (B) |
|--------------------|---------------|---------------|-------------------------------|--------------------|
| dist/index.mjs     | 26.11         | 25.93                                  | 7.69               | 7.63                                              |
| dist/index.umd.js  | 18.46         | 18.47                                  | 6.53               | 6.53                                              |

Bundlesize aside, it gets harder to maintain all the newly added regions (in the future) unless we centralize our effort. Let me know what you think! 🙂

**Review by ts-storyblok — APPROVED** [review]

**Review by ademarCardoso — APPROVED** [review]
My concern about dependency still remains, but I think we can go down that road and test it, thanks for the feedback @eunjae-lee . Good work on the region helper and here 🚀

**Comment by eunjae-lee** [comment]
Thanks @ademarCardoso !

Then what's the next step? Will a QA be involved here? Or are we merging this? Let me know if I need to do anything :)

**Comment by BibiSebi** [comment]
@eunjae-lee after looking a bit at the github actions, i believe they will handle the release once this PR is merged to main.

**Comment by alvarosabu** [comment]
Close due inactivity
